src/model/paths_types.ml

```diff
@@ -259,6 +262,8 @@ and Resolved_fragment : sig
     | `Class of signature * ClassName.t
     | `ClassType of signature * ClassTypeName.t ]
 
+  type leaf = [ module_ | type_ ]
```

Comment:
What about classes, substs and others ?
(Would help move the `assert false` from `render_resolved_fragment`)

Reply:
this refers to the type of the substitution - ie, `X with module Y=Z` or `X with type t = u` -- so only module or type required.

Reply:
Thanks, I got confused by the `type t` below. Why does it include everything ?

Reply:
that's the union one, helpful for printers and the like. it's quite handy to have just the leaf one so we don't have to deal with the Root constructor unnecessarily